File: app/src/main/java/com/kkkcut/e20j/ui/fragment/engraving/TemplateSelectFragment.java
package com.kkkcut.e20j.ui.fragment.engraving;

import android.os.Bundle;
import android.text.Editable;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.OnClick;
import butterknife.OnTextChanged;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.kkkcut.e20j.DbBean.userDB.KeyMarkingTemplate;
import com.kkkcut.e20j.adapter.KeyMarkingTemplateAdapter;
import com.kkkcut.e20j.dao.UserDataDaoManager;
import com.kkkcut.e20j.ui.dialog.RemindDialog;
import com.kkkcut.e20j.ui.fragment.BaseBackFragment;
import com.kkkcut.e20j.us.R;
import io.reactivex.Observable;
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;
import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.Callable;

/* loaded from: classes.dex */
public class TemplateSelectFragment extends BaseBackFragment implements BaseQuickAdapter.OnItemChildClickListener, BaseQuickAdapter.OnItemClickListener {

    @BindView(R.id.bt_delete_all)
    Button btDeleteAll;

    @BindView(R.id.et_search)
    EditText etSearch;
    private KeyMarkingTemplateAdapter keyMarkingTemplateAdapter;

    @BindView(R.id.rv_template_list)
    RecyclerView rvTemplateList;

    @Override // com.kkkcut.e20j.androidquick.ui.base.QuickFragment
    protected int getContentViewLayoutID() {
        return R.layout.fragment_template_list;
    }

    public static TemplateSelectFragment newInstance() {
        Bundle bundle = new Bundle();
        TemplateSelectFragment templateSelectFragment = new TemplateSelectFragment();
        templateSelectFragment.setArguments(bundle);
        return templateSelectFragment;
    }

    @Override // com.kkkcut.e20j.ui.fragment.BaseBackFragment
    public String setTitleStr() {
        return getString(R.string.template_selection);
    }

    @Override // com.kkkcut.e20j.androidquick.ui.base.QuickFragment
    protected void initViewsAndEvents() {
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
        linearLayoutManager.setOrientation(1);
        this.rvTemplateList.setLayoutManager(linearLayoutManager);
        KeyMarkingTemplateAdapter keyMarkingTemplateAdapter = new KeyMarkingTemplateAdapter();
        this.keyMarkingTemplateAdapter = keyMarkingTemplateAdapter;
        keyMarkingTemplateAdapter.setOnItemChildClickListener(this);
        this.keyMarkingTemplateAdapter.setOnItemClickListener(this);
        this.rvTemplateList.setAdapter(this.keyMarkingTemplateAdapter);
        getTemplates();
    }

    private void getTemplates() {
        addDisposable(Observable.fromCallable(new Callable<List<KeyMarkingTemplate>>() { // from class: com.kkkcut.e20j.ui.fragment.engraving.TemplateSelectFragment.3
            @Override // java.util.concurrent.Callable
            public List<KeyMarkingTemplate> call() throws Exception {
                return UserDataDaoManager.getInstance(TemplateSelectFragment.this.getContext()).getKeyMarkingTemplates();
            }
        }).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(new Consumer<List<KeyMarkingTemplate>>() { // from class: com.kkkcut.e20j.ui.fragment.engraving.TemplateSelectFragment.1
            @Override // io.reactivex.functions.Consumer
            public void accept(List<KeyMarkingTemplate> list) throws Exception {
                TemplateSelectFragment.this.keyMarkingTemplateAdapter.setNewData(list);
            }
        }, new Consumer<Throwable>() { // from class: com.kkkcut.e20j.ui.fragment.engraving.TemplateSelectFragment.2
            @Override // io.reactivex.functions.Consumer
            public void accept(Throwable th) throws Exception {
            }
        }));
    }

    @OnClick({R.id.bt_delete_all})
    public void onViewClicked() {
        RemindDialog remindDialog = new RemindDialog(getContext());
        remindDialog.setRemindMsg(getString(R.string.delete_all_recordes));
        remindDialog.setDialogBtnCallback(new RemindDialog.DialogBtnCallBack() { // from class: com.kkkcut.e20j.ui.fragment.engraving.TemplateSelectFragment.4
            @Override // com.kkkcut.e20j.ui.dialog.RemindDialog.DialogBtnCallBack
            public void onDialogButClick(boolean z) {
                if (z) {
                    TemplateSelectFragment.this.keyMarkingTemplateAdapter.setNewData(new ArrayList());
                    UserDataDaoManager.getInstance(TemplateSelectFragment.this.getContext()).deleteAllTemplate();
                }
            }
        });
        remindDialog.show();
    }

    @Override // com.chad.library.adapter.base.BaseQuickAdapter.OnItemChildClickListener
    public void onItemChildClick(BaseQuickAdapter baseQuickAdapter, View view, int i) {
        UserDataDaoManager.getInstance(getContext()).deleteSingleTemplate((KeyMarkingTemplate) baseQuickAdapter.getData().get(i));
        baseQuickAdapter.remove(i);
    }

    /* JADX INFO: Access modifiers changed from: package-private */
    @OnTextChanged(callback = OnTextChanged.Callback.AFTER_TEXT_CHANGED, value = {R.id.et_search})
    public void afterTextChanged(final Editable editable) {
        Disposable subscribe = Observable.fromCallable(new Callable<List<KeyMarkingTemplate>>() { // from class: com.kkkcut.e20j.ui.fragment.engraving.TemplateSelectFragment.6
            @Override // java.util.concurrent.Callable
            public List<KeyMarkingTemplate> call() throws Exception {
                return UserDataDaoManager.getInstance(TemplateSelectFragment.this.getContext()).fuzzyQueryTemplates(editable.toString());
            }
        }).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(new Consumer<List<KeyMarkingTemplate>>() { // from class: com.kkkcut.e20j.ui.fragment.engraving.TemplateSelectFragment.5
            @Override // io.reactivex.functions.Consumer
            public void accept(List<KeyMarkingTemplate> list) throws Exception {
                TemplateSelectFragment.this.keyMarkingTemplateAdapter.setNewData(list);
            }
        });
        clearDisposable();
        addDisposable(subscribe);
    }

    @Override // com.chad.library.adapter.base.BaseQuickAdapter.OnItemClickListener
    public void onItemClick(BaseQuickAdapter baseQuickAdapter, View view, int i) {
        Bundle bundle = new Bundle();
        bundle.putSerializable("template", (KeyMarkingTemplate) baseQuickAdapter.getData().get(i));
        setFragmentResult(0, bundle);
        onBack();
    }
}

File: app/src/main/java/com/kkkcut/e20j/ui/fragment/engraving/TemplateSelectFragment.kt
package com.kkkcut.e20j.ui.fragment.engraving

import android.os.Bundle
import android.text.Editable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.chad.library.adapter.base.BaseQuickAdapter
import com.kkkcut.e20j.DbBean.userDB.KeyMarkingTemplate
import com.kkkcut.e20j.adapter.KeyMarkingTemplateAdapter
import com.kkkcut.e20j.dao.UserDataDaoManager
import com.kkkcut.e20j.ui.dialog.RemindDialog
import com.kkkcut.e20j.ui.fragment.BaseBackFragment
import com.kkkcut.e20j.us.R
import com.kkkcut.e20j.us.databinding.FragmentTemplateListBinding
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.disposables.Disposable
import io.reactivex.rxjava3.schedulers.Schedulers

/* loaded from: classes.dex */
class TemplateSelectFragment : BaseBackFragment(), BaseQuickAdapter.OnItemChildClickListener,
    BaseQuickAdapter.OnItemClickListener {
    var binding: FragmentTemplateListBinding? = null

    private var keyMarkingTemplateAdapter: KeyMarkingTemplateAdapter? = null

    override fun onCreateView(
        layoutInflater: LayoutInflater,
        viewGroup: ViewGroup?,
        bundle: Bundle?
    ): View {
        super.onCreateView(layoutInflater, viewGroup, bundle)
        binding = FragmentTemplateListBinding.inflate(layoutInflater, viewGroup, false)
        return binding!!.getRoot()
    }

    // com.kkkcut.e20j.androidquick.ui.base.QuickFragment
    override fun getContentViewLayoutID(): Int {
        return R.layout.fragment_template_list
    }

    // com.kkkcut.e20j.ui.fragment.BaseBackFragment
    override fun setTitleStr(): String {
        return getString(R.string.template_selection)
    }

    // com.kkkcut.e20j.androidquick.ui.base.QuickFragment
    override fun initViewsAndEvents() {
        val linearLayoutManager = LinearLayoutManager(requireContext())
        linearLayoutManager.setOrientation(1)
        binding!!.rvTemplateList.setLayoutManager(linearLayoutManager)

        this.keyMarkingTemplateAdapter = KeyMarkingTemplateAdapter()
        keyMarkingTemplateAdapter!!.onItemChildClickListener = this
        keyMarkingTemplateAdapter!!.onItemClickListener = this
        binding!!.rvTemplateList.setAdapter(this.keyMarkingTemplateAdapter)
        templates
    }

    private val templates: Unit
        get() {
            addDisposable(Observable.fromCallable {
                UserDataDaoManager.getInstance(this.requireContext())
                    .keyMarkingTemplates
            }.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(
                { list: List<KeyMarkingTemplate> ->
                    keyMarkingTemplateAdapter!!.setNewData(
                        list
                    )
                }, { }, { dismissLoadingDialog() }
            )
            )
        }

    fun onViewClicked() {
        val remindDialog = RemindDialog(requireContext())
        remindDialog.setRemindMsg(getString(R.string.delete_all_recordes))
        remindDialog.setDialogBtnCallback { z: Boolean ->
            if (z) {
                keyMarkingTemplateAdapter!!.setNewData(ArrayList())
                UserDataDaoManager.getInstance(this.requireContext())
                    .deleteAllTemplate()
            }
        }
        remindDialog.show()
    }

    /* JADX INFO: Access modifiers changed from: package-private */
    fun afterTextChanged(editable: Editable) {
        val subscribe: Disposable = Observable.fromCallable {
            UserDataDaoManager.getInstance(this.requireContext()).fuzzyQueryTemplates(editable.toString())
        }.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe({ list: List<KeyMarkingTemplate> ->
            keyMarkingTemplateAdapter!!.setNewData(list)
        }, { dismissLoadingDialog() })
        clearDisposable()
        addDisposable(subscribe)
    }

    override fun onItemClick(adapter: BaseQuickAdapter<*, *>?, view: View, position: Int) {
    }

    override fun onItemChildClick(adapter: BaseQuickAdapter<*, *>, view: View, position: Int) {
        val bundle = Bundle()
        bundle.putSerializable("template", adapter.data[position] as KeyMarkingTemplate?)
        setFragmentResult(0, bundle)
        onBack()
    }

    companion object {
        fun newInstance(): TemplateSelectFragment {
            val bundle = Bundle()
            val templateSelectFragment = TemplateSelectFragment()
            templateSelectFragment.setArguments(bundle)
            return templateSelectFragment
        }
    }
}
